upgrade-instructions.md

```diff
@@ -17,6 +17,41 @@ Changes to apply between 3.3.x and 3.4.x will be reported here.
 Change list:
 
  - [Upgrade to ElasticSearch 6.3](#upgrade-to-elasticsearch-6.3)
+ 
+#### Enqueuing several time a mail with the same name
+
+Date: XX/06/2019
+
+SHA-1: XXXXX
+
+JIRA: https://issues.apache.org/jira/browse/JAMES-2794
+
+Concerned products: (experimental) RabbitMQ MailQueue
+
+RabbitMQ mail queue combines RabbitMQ with projections in Cassandra to offer advanced management capabilities expected from
+a mail queue (browse, delete, size, clear). In these projections, the mails are identified by there name. Thus enqueuing a
+mail that had already been processed will lead the given email to be considered already deleted and it will be discarded 
+and lost.
+
+This is an issue, as several other components build features around submitting a mail several time with the name. 
+
+For instance:
+
+ - MailRepository reprocessing
+ - RemoteDelivery bouncing under some configurations
+ - RecipientRewriteTable rewriting to a remote server
+
+We thus changed the table structure of RabbitMQ mail queue projections to be built around an EnqueueId. This additional 
+level of indirection allows several enqueues with the same name.
+
+Upgrade to the newest James server needs to be performed with an empty MailQueue.
+
+To do so:
+
+ - Given a distributed Guice James server
+ - Stop incoming traffic
```

Comment:
easier said than done

Reply:
Using docker you can restart with no exposed ports. In k8s I understood this was doable. Otherwise a firewall rule does the trick.

Reply:
(agree that we could expose webadmin endpoints for stopping given protocols.)

Reply:
It's not what I have in mind, stop a mail server doesn't make sense because you'll stop receiving emails ... I think we don't care for now but it's something that we can't ask people to do in case of migration.

Reply:
I know, I considered it acceptable as it is considered "not supported".

An alternative would be to run the two version concurrently, with different Rabbit instances but common Rabbit EventBus. Much harder - I saved some pain here to be honest.

Reply:
> you'll stop receiving emails

not exactly, you should just delay them as smtp server as supposed to retry a defined amount of time before bouncing, no?

Reply:
Yes, but I don't think you want to rely on that. However, I think there's a tactic using a backup MX for that.
We'll need to explore that in the future. Let's close the debate for now.